src/main/java/dev/pgm/community/party/MapPartyCommands.java

```diff
@@ -67,163 +69,151 @@ public void createPreset(CommandAudience viewer, Player sender, String presetNam
-                  return;
-                }
-                hosts.addSubHost(player);
-              });
-    }
+  public void viewHosts(CommandAudience viewer, Player sender) {
+    if (isPartyMissing(viewer)) return;
+    new HostMenu(party, sender);
+  }
 
-    @Subcommand("remove")
-    @Syntax("[player]")
-    @CommandCompletion("@players") // TODO: get cached offline names for completion here
-    public void removeHost(CommandAudience viewer, String target) {
-      if (isPartyMissing(viewer)) return;
-      MapPartyHosts hosts = party.getParty().getHosts();
-      if (!hosts.removeSubHost(target)) {
-        viewer.sendWarning(
-            text()
-                .append(text(target, NamedTextColor.DARK_AQUA))
-                .append(text(" is not a party host"))
-                .build());
-      }
+  @CommandMethod(HOST_CMD + " add <targets>")
+  @CommandPermission(CommunityPermissions.PARTY)
+  public void addHost(CommandAudience viewer, @Argument("targets") String targets) {
+    if (isPartyMissing(viewer)) return;
+    MapPartyHosts hosts = party.getParty().getHosts();
+    PlayerSelection selection = getPlayers(viewer, targets);
+    selection
+        .getPlayers()
+        .forEach(
+            player -> {
+              if (!party.canHost(player)) {
+                viewer.sendWarning(MapPartyMessages.getAddHostError(player));
+                return;
+              }
+
+              if (hosts.isHost(player.getUniqueId())) {
+                viewer.sendWarning(MapPartyMessages.getExistingHostError(player));
+                return;
+              }
+              hosts.addSubHost(player);
+            });
+  }
+
+  @CommandMethod(HOST_CMD + " remove <target>")
+  @CommandPermission(CommunityPermissions.PARTY)
+  public void removeHost(CommandAudience viewer, @Argument("target") String target) {
+    if (isPartyMissing(viewer)) return;
+    MapPartyHosts hosts = party.getParty().getHosts();
+    if (!hosts.removeSubHost(target)) {
+      viewer.sendWarning(
+          text()
+              .append(text(target, NamedTextColor.DARK_AQUA))
+              .append(text(" is not a party host"))
+              .build());
     }
+  }
 
-    @Subcommand("transfer")
-    @Syntax("[player]")
-    @CommandCompletion("@players")
-    public void transferHost(CommandAudience viewer, String target) {
-      if (isPartyMissing(viewer)) return;
-
-      MapPartyHosts hosts = party.getParty().getHosts();
-      Player player = getSinglePlayer(viewer, target, true);
+  @CommandMethod(HOST_CMD + " transfer <target>")
+  @CommandPermission(CommunityPermissions.PARTY)
+  public void transferHost(CommandAudience viewer, @Argument("target") String target) {
+    if (isPartyMissing(viewer)) return;
 
-      if (!party.canHost(player)) {
-        viewer.sendWarning(MapPartyMessages.getAddHostError(player));
-        return;
-      }
+    MapPartyHosts hosts = party.getParty().getHosts();
+    Player player = getSinglePlayer(viewer, target, true);
```

Comment:
things like these should be swapped over to use parsers, instead of String argument, use a Player argument directly
It should handle all of the if-you-can-see-or-not thing, and the nick, etc

Reply:
@Pablete1234 Great idea, I'm starting to go through all commands and see where we can use parsers instead.

Though there's one area I'd appreciate some advice in:

How can we effectively handle commands that target either online or offline players, where the name of the player can be auto-completed if they are online, but still accepted (returns the input of the name) even if they are offline? For example, if a staff member performs the command `/seen applenick`, and I am online, my name should be able to be tab-completed. However, the command should also be able to accept the name of an offline player, such as `/seen Pablete1234`, if you were offline. 

Would creating a custom player object and parser be the solution? If you have any alternative suggestions would be really helpful. Thanks!

```java
  @CommandMethod("seen|lastseen|find <target>")
  @CommandDescription("View when a player was last online")
  @CommandPermission(CommunityPermissions.FIND)
  public void seenPlayer(CommandAudience audience, @Argument("target") String target) {
```

Reply:
I believe Player extends OfflinePlayer, so that's an abstraction that can work (ie: you have an OfflinePlayer parameter, that doesn't mean the Player object you get can't be an actual (online) `Player` object)